tests/spanner_dbapi/test_parse_utils.py

```diff
@@ -337,112 +338,40 @@ def test_sql_pyformat_args_to_spanner_invalid(self):
-                    "a1": 10,
-                    "b1": TimestampStr("2005-08-30T01:01:01.000001Z"),
-                    "c1": DateStr("2005-08-30"),
-                },
-                {
-                    "a1": param_types.INT64,
-                    "b1": param_types.TIMESTAMP,
-                    "c1": param_types.DATE,
-                },
-            ),
-            (
-                {"a1": 10, "b1": "aaaaa08-30T01:01:01.000001Z"},
-                {"a1": param_types.INT64, "b1": param_types.STRING},
-            ),
-            (
-                {
-                    "a1": 10,
-                    "b1": "2005-08-30T01:01:01.000001",
-                    "t1": True,
-                    "t2": False,
-                    "f1": 99e9,
-                },
-                {
-                    "a1": param_types.INT64,
-                    "b1": param_types.STRING,
-                    "t1": param_types.BOOL,
-                    "t2": param_types.BOOL,
-                    "f1": param_types.FLOAT64,
-                },
-            ),
-            (
-                {"a1": 10, "b1": "2019-11-26T02:55:41.000000Z"},
-                {"a1": param_types.INT64, "b1": param_types.STRING},
-            ),
-            (
-                {
-                    "a1": 10,
-                    "b1": TimestampStr("2019-11-26T02:55:41.000000Z"),
-                    "dt1": datetime.datetime(2011, 9, 1, 13, 20, 30),
-                    "d1": datetime.date(2011, 9, 1),
-                },
-                {
-                    "a1": param_types.INT64,
-                    "b1": param_types.TIMESTAMP,
-                    "dt1": param_types.TIMESTAMP,
-                    "d1": param_types.DATE,
-                },
-            ),
-            (
-                {"a1": 10, "b1": TimestampStr("2019-11-26T02:55:41.000000Z")},
-                {"a1": param_types.INT64, "b1": param_types.TIMESTAMP},
-            ),
-            ({"a1": b"bytes"}, {"a1": param_types.BYTES}),
-            ({"a1": 10, "b1": None}, {"a1": param_types.INT64}),
-            (None, None),
-        ]
+        params = {
+            "a1": 10,
+            "b1": "string",
+            "c1": 10.39,
+            "d1": TimestampStr("2005-08-30T01:01:01.000001Z"),
+            "e1": DateStr("2019-12-05"),
+            "f1": True,
+            "g1": datetime.datetime(2011, 9, 1, 13, 20, 30),
+            "h1": datetime.date(2011, 9, 1),
+            "i1": b"bytes",
+            "j1": None,
+        }
+        want_types = {
+            "a1": param_types.INT64,
+            "b1": param_types.STRING,
+            "c1": param_types.FLOAT64,
+            "d1": param_types.TIMESTAMP,
+            "e1": param_types.DATE,
+            "f1": param_types.BOOL,
+            "g1": param_types.TIMESTAMP,
+            "h1": param_types.DATE,
+            "i1": param_types.BYTES,
+        }
+        got_types = get_param_types(params)
+        self.assertEqual(got_types, want_types)
```

Comment:
Don't know why so many test cases were written - they are covering the same cases. Shortened them